### DISCUSSION TITLE
Animation retargeting 3d

### DISCUSSION THREAD

**Discussion by TomerconDevelopers** [axmolengine/axmol#1183] [Q&A]
**Is your feature request related to a problem? Please describe.**
I want to retarget humanoid animations from one model to another

**Describe the solution you'd like**
I don't know how to do it

**Describe alternatives you've considered**
Nothing

**Additional context**

**Comment by rh101** [comment]
From what little information you have provided, this is not an issue or a feature request, and should be placed in "Discussions".

**Comment by aismann** [comment]
@TomerconDevelopers 
Is this a theoretical question or having you a non working example code/snippet or any others using axmol?

**Reply by TomerconDevelopers** [reply]
I will describe the feature clearly,

Suppose i have a 3d model having animations walk, run, jump etc..
And i have another 3d model with no animations. Animation retargeting is the process of playing animations of one 3d model in model 2 if both model sharing same skeleton structure (humanoid ) directly in engine without using any external tools.


Why this important?

My game having 10 type of different playable characters and 100+ different enemies, all are having humanoid skeleton. Animating all models individually in an animation editor is difficult task.

**Reply by aismann** [reply]
> I will describe the feature clearly,
> 
> Suppose i have a 3d model having animations walk, run, jump etc.. And i have another 3d model with no animations. Animation retargeting is the process of playing animations of one 3d model in model 2 if both model sharing same skeleton structure (humanoid ) directly in engine without using any external tools.
> 
> Why this important?
> 
> My game having 10 type of different playable characters and 100+ different enemies, all are having humanoid skeleton. Animating all models individually in an animation editor is difficult task.

Ok. But why you ask here?
Its not realy an axmol issue.
Maybe there be tools which can do your feature wish out of the box?

It is not better to ask on a 3D forum like blender or so?

**Reply by rh101** [reply]
@TomerconDevelopers If there are no samples in cpp-tests that do something similar, and no-one else knows how this is done, or even if it's possible, then the best thing you can do is just review the Axmol source code related to the 3D animations and try to work it out yourself.

**Reply by aismann** [reply]
> ...l are having humanoid skeleton. Animating all models individually in an animation editor is difficult task.

Looks like you using Spine or any other animation/bones tools.
Better you ask on this forums.

**Reply by solan-solan** [reply]
If to say about 3d, you could try Omniverse for your purpose:
https://docs.omniverse.nvidia.com/app_machinima/prod_extensions/ext_animation-retargeting.html
https://actorcore.reallusion.com/learn-and-support/my-software/omniverse/step-by-step-guide/motion-retargeting

Blender has many addons also. Paid are better, but it could be found free:
https://github.com/Mwni/blender-animation-retargeting

Unreal engine has integrated retarget manager, but may be it allows only to retarget from its mannequin to other skeletals
https://bearhammergames.com/2021/10/26/tutorial-retargeting-ue4-mannequin-animations-onto-a-nonstandard-bipedal-skeleton/
